cmd/cluster-agent/api/agent/agent.go

```diff
@@ -159,10 +164,45 @@ func getConfigCheck(w http.ResponseWriter, r *http.Request) {
 	jsonConfig, err := json.Marshal(response)
 	if err != nil {
 		log.Errorf("Unable to marshal config check response: %s", err)
-		body, _ := json.Marshal(map[string]string{"error": err.Error()})
-		http.Error(w, string(body), 500)
+		setJSONError(w, err, 500)
 		return
 	}
 
 	w.Write(jsonConfig)
 }
+
+func getTaggerList(w http.ResponseWriter, r *http.Request) {
+	response := tagger.List(collectors.HighCardinality)
+
+	jsonTags, err := json.Marshal(response)
+	if err != nil {
+		setJSONError(w, log.Errorf("Unable to marshal tagger list response: %s", err), 500)
+		return
+	}
+	w.Write(jsonTags)
+}
+
+func getVerboseWorkloadList(w http.ResponseWriter, r *http.Request) {
+	workloadList(w, true)
+}
+
+func getShortWorkloadList(w http.ResponseWriter, r *http.Request) {
+	workloadList(w, false)
+}
+
+func workloadList(w http.ResponseWriter, verbose bool) {
+	response := workloadmeta.GetGlobalStore().Dump(verbose)
+	jsonDump, err := json.Marshal(response)
+	if err != nil {
+		setJSONError(w, log.Errorf("Unable to marshal workload list response: %v", err), 500)
+		return
+	}
+
+	w.Write(jsonDump)
+}
```

Comment:
Following my previous suggestion of passing the verbosity as a parameter:

```suggestion
func workloadList(w http.ResponseWriter, r *http.Request) {
    verbose := false
    params := r.URL.Query()
    if v, ok := params["verbose"]; ok {
      if len(v) >= 1 && v[0] == "true" {
        verbose = true
        }
    }
    
	response := workloadmeta.GetGlobalStore().Dump(verbose)
	jsonDump, err := json.Marshal(response)
	if err != nil {
		setJSONError(w, log.Errorf("Unable to marshal workload list response: %v", err), 500)
		return
	}

	w.Write(jsonDump)
}
```